3.7/graphs-satellite-graphs.md

```diff
@@ -0,0 +1,134 @@
+a cluster, which enables DB-Servers to execute parts of joins locally.
+
+This greatly improves performance for such joins.
+
+To create a satellite graph in arangosh, use the `satelliteGraph` module.
+
+    arangosh> satGM = require("@arangodb/satellite-graph")
+    arangosh> satGM._create("satelliteGraph", [ satGM._relation("edges", "vertices", "vertices") ], [], {});
+
+You can also create vertex and edge collections manually, and set their `replicationFactor` to
+`satellite`
+
+    arangosh> db._createDocumentCollection("vertices2", { replicationFactor: "satellite" });
+    arangosh> db._createEdgeCollection("edges2", { replicationFactor: "satellite" });
+
+A full example
+--------------
+
+    arangosh> var explain = require("@arangodb/aql/explainer").explain
+    arangosh> graphModule = require("@arangodb/general-graph");
+    arangosh> satGraphModule = require("@arangodb/satellite-graph");
+    arangosh> graphModule._create("normalGraph", [ graphModule._relation("edges", "vertices", "vertices") ], [], {});
+    arangosh> satGraphModule._create("satelliteGraph", [ satGM._relation("satEdges", "satVertices", "satVertices") ], [], {});
+    arangosh> db._create("collection", {numberOfShards: 8})
+    arangosh> db._create("collection2", {numberOfShards: 8})
+
+Let's analyse a join involving a traversal.
+
+```
+arangosh> explain(`FOR doc in collection FOR v,e,p IN OUTBOUND "vertices/start" GRAPH "normalGraph" RETURN [doc,v,e,p]`)
+
+Query String (99 chars, cacheable: true):
+ FOR doc in collection FOR v,e,p IN OUTBOUND "vertices/start" GRAPH "normalGraph" RETURN [doc,v,e,p]
+
+Execution plan:
+ Id   NodeType                  Site  Est.   Comment
+  1   SingletonNode             DBS      1   * ROOT
+  2   EnumerateCollectionNode   DBS      0     - FOR doc IN collection   /* full collection scan, 8 shard(s) */
+  8   RemoteNode                COOR     0       - REMOTE
+  9   GatherNode                COOR     0       - GATHER 
+  3   TraversalNode             COOR     0       - FOR v  /* vertex */, e  /* edge */, p  /* paths */ IN 1..1  /* min..maxPathDepth */ OUTBOUND 'vertices/start' /* startnode */  GRAPH 'normalGraph'
+  4   CalculationNode           COOR     0         - LET #6 = [ doc, v, e, p ]   /* simple expression */   /* collections used: doc : collection */
+  5   ReturnNode                COOR     0         - RETURN #6
+
+Indexes used:
+ By   Name   Type   Collection   Unique   Sparse   Selectivity   Fields        Ranges
+  3   edge   edge   edges        false    false       100.00 %   [ `_from` ]   base OUTBOUND
+
+Traversals on graphs:
+ Id  Depth  Vertex collections  Edge collections  Options                                  Filter / Prune Conditions
+ 3   1..1   vertices            edges             uniqueVertices: none, uniqueEdges: path                           
+
+Optimization rules applied:
+ Id   RuleName
+  1   scatter-in-cluster
+  2   remove-unnecessary-remote-scatter
+```
```

Comment:
I think it's possible to specify the commands and let the output be generated with the documentation. But its been a long time since I've used that last, so it might have changed. @Simran-B, can you shed some light on that?